### PR TITLE
Get rid of StandardOutput=syslog

### PR DESCRIPTION
**Pull request by poettering**
It's confusing and mostly a historic artifact. let's patch to StandardOutput=journal automatically, and log about it.

Fixes: #15807

### CODE REVIEW

**Comment by mbiebl** [comment]
http://codesearch.debian.net/search?q=StandardOutput%3Dsyslog&literal=1

**Comment by mbiebl** [comment]
Does this also affect `StandardError=`?

**Comment by poettering** [comment]
i think early howtos suggested that.

yes, it applies to both the stdout and the stderr stuff

File: src/core/load-fragment.c [hunk 1]
```diff
@@ -1154,6 +1163,11 @@ int config_parse_exec_output(
                 }
         }
 
+        if (obsolete)
+                log_syntax(unit, LOG_WARNING, filename, line, 0,
```
Comment (keszybz) [inline src/core/load-fragment.c]: I think this should be LOG_NOTICE at most. With the amount of matches that @mbiebl posted, LOG_WARNING seems excessive.

Comment (poettering) [reply src/core/load-fragment.c]: good point. in tmpfiles when we patch /var/run/ to /run/ we also use LOG_NOTICE:

https://github.com/systemd/systemd/blob/master/src/tmpfiles/tmpfiles.c#L2482

**Comment by poettering** [comment]
force pushed new version. Only change is the LOG_NOTICE change. Took liberty to upgrade green label.

**Comment by mbiebl** [comment]
Could you update the commit message to mention `StandardError=`?

I guess we could add a lintian check in Debian which checks for this. 
What would be the recommended action to resolve this?
Drop StandardOutput/StandardError or replace it? Something else?

Would be great if there was a recommendation what should be used instead.

**Comment by poettering** [comment]
> Would be great if there was a recommendation what should be used instead.

The log message makes a suggestion already: "Please update your unit file, and consider removing the setting altogether." 

That should be enough, no?

**Comment by poettering** [comment]
added this StandardError= thing to the commit msg

**Comment by mbiebl** [comment]
> > Would be great if there was a recommendation what should be used instead.
> 
> The log message makes a suggestion already: "Please update your unit file, and consider removing the setting altogether."
> 
> That should be enough, no?

Yes, thanks.
I'll try to get lintian updated accordingly and issue a warning.

**Comment by keszybz** [comment]
bionic-i386 timed out.

**Pull request merged**